presto-tests/src/test/java/com/facebook/presto/tests/TestLegacyLogFunction.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.facebook.presto.testing.MaterializedResult;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static io.airlift.testing.Assertions.assertContains;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
+
+public class TestLegacyLogFunction
+{
+    private static final String QUERY = "SELECT LOG(25, 5)";
+
+    @Test
+    public void testLegacyLogFunctionEnabled()
+            throws Exception
+    {
+        try (DistributedQueryRunner queryRunner = createQueryRunner(true)) {
+            MaterializedResult result = queryRunner.execute(QUERY);
+            assertEquals(result.getOnlyValue(), 2.0);
+        }
+    }
+
+    @Test
+    public void testLegacyLogFunctionDisabled()
+    {
+        try (DistributedQueryRunner queryRunner = createQueryRunner(false)) {
+            queryRunner.execute(QUERY);
+            fail("Expected exception");
+        }
+        catch (Exception e) {
+            assertContains(e.getMessage(), "Function log not registered");
```

Comment:
Since your try catch spans the whole test, it'd better with `@Test(expectedException=..., expectedExceptionMsg=..)`

However better yet would be to wrap just `queryRunner.execute(QUERY);` with a try-catch